src/Documents.jl

```diff
@@ -223,6 +220,8 @@ struct Internal
     errors::Set{Symbol}
 end
 
+abstract type DocumenterPlugin end
```

Comment:
This should probably move into the `Documenter` module.

Also, I wonder if it might be better to call it just `Plugin`. In this case, it shouldn't be exported, but it will be part of public API. The users should access it with `Documenter.Plugin`.